templates/vm.xml.j2

```diff
@@ -39,9 +39,9 @@
       <source pool='{{ volume.pool }}' volume='{{ volume.name }}'/>
       {% endif %}
       {% if volume.target is undefined %}
-      <target dev='vd{{ 'abcdefghijklmnopqrstuvwxyz'[loop.index - 1] }}'/>
+      <target dev='vd{{ 'abcdefghijklmnopqrstuvwxyz'[loop.index - 1] }}' {% if volume.device | default(libvirt_volume_default_device) == 'cdrom' %}bus='sata'{% else %}bus='virtio'{% endif %}/>
```

Comment:
Is there a reason why we need to specify the bus here? Previously we have left it undefined and libvirt selects one for us.

Reply:
There was a change in libvirt that now requires to specify a bus if it's a cdrom. I was having errors with the latest version of libvirt without specifying a bus. Can't find the change yet.

Reply:
Ok, so, this is what happens without setting the bus:
```
libvirt.libvirtError: unsupported configuration: disk type 'virtio' of 'vdb' does not support ejectable media
```

this was the generated XML:
```
    <disk type='volume' device='disk'>
      <driver name='qemu' type='raw'/>
        <source pool='vm-storage' volume='archboot3'/>
        <target dev='vda'/>
      </disk>
    <disk type='file' device='cdrom'>
      <driver name='qemu' type='raw'/>
      <source file='/var/lib/libvirt/images/archboot3.iso'/>
      <target dev='vdb'/>
    </disk>
```

this is how I defined the vm volumes:
```
            volumes:
              - name: archboot3
                type: volume
                device: disk
                pool: vm-storage
                format: raw
                capacity: 50G
              - name: archboot3.iso              
                type: file
                device: cdrom
                capacity: 5G
                format: raw
                image: https://mirror.fsmg.org.nz/archlinux/iso/archboot/2018.06/archlinux-2018.06-1-archboot-network.iso
```


Looking at this line in the source code of libvirt, which has the error message:
https://github.com/libvirt/libvirt/blob/master/src/qemu/qemu_domain.c#L6936 

We can see, that if the device is `cdrom` and the bus is `virtio` (Which, I guess, is the bus used by default) - libvirt throws that error message.

BTW, this is XML generated by virt-manager:
```
<disk type="file" device="cdrom">
  <driver name="qemu" type="raw"/>
  <source file="/var/lib/libvirt/images/iso/archlinux-2018.06-1-archboot-network.iso"/>
  <target dev="sda" bus="sata"/>
  <readonly/>
</disk>
```

Again, notice that the bus is specifically set to `sata`.


When I was searching for a solution to this, I found a conversation between libvirt devs, that cdrom type device shouldn't accept virtio bus, and that was accepted before. They fixed the behaviour, by making this change mandatory. Unfortunately, I can't find the discussion.

Reply:
Still can't find the discussion, but this is where the change was made: https://libvirt.org/git/?p=libvirt.git;a=commit;h=5d884f3d3c3b0639ce06152f3f91d6ac87d13cb6

Reply:
Thanks for the detailed follow up. I think if we remove the else here to make it the same as the other one we're good to go.